### ISSUE TITLE
The kl_loss increases fast

### ISSUE THREAD

**Issue by gauss-clb** [agentica-project/rllm#94]
When I use verl to train deepseek-r1-distill-qwen-1.5b, the train reward and kl_loss increases fast within 10 steps. And after 10 steps, the train reward starts to enter into plateaus and eventually declines .  Do you have any idea about why the process only contain dozens of steps.

![Image](https://github.com/user-attachments/assets/5c1399c6-a263-4489-9008-e50e73fdc7b3)

![Image](https://github.com/user-attachments/assets/9adc231c-6d91-4ad8-b891-208c34fbb6c7)

![Image](https://github.com/user-attachments/assets/4c8a9c59-8a4a-4d61-8f75-451079c597dd)

**Comment by michaelzhiluo** [comment]
Which dataset are you using? It should be relatively stable on the Deepscaler Dataset (also suggest using DAPO-related parameters such as `clip_low`, `clip_high` and `mask_truncated_samples` to further stabilize training)

**Comment by gauss-clb** [comment]
@michaelzhiluo This happened when I used original verl framework and the data format like,
```
system_prompt:
Your role as an assistant involves thoroughly exploring questions through a systematic long thinking process before providing the final precise and accurate solutions. This requires engaging in a comprehensive cycle of analysis, summarizing, exploration, reassessment, reflection, backtracing, and iteration to develop well-considered thinking process. Please structure your response into two main sections: Thought and Solution. In the Thought section, detail your reasoning process using the specified format: <think> {thought with steps separated with '\n\n'} </think> Each step should include detailed considerations such as analisying questions, summarizing relevant findings, brainstorming new ideas, verifying the accuracy of the current steps, refining any errors, and revisiting previous steps. In the Solution section, based on various attempts, explorations, and reflections from the Thought section, systematically present the final solution that you deem correct. The solution should remain a logical, accurate, concise expression style and detail necessary step needed to reach the conclusion. Now, try to solve the following question through the above guidelines, please use the same language as the following question.

user_prompt:
Return your final response within \\boxed{}.\n
{math_question}
```

And I also tried rllm and use the data format above(the blue curve), your data format(the gray curve). Do you think adding a system prompt is a problem?

![Image](https://github.com/user-attachments/assets/1a653f44-8b89-4ad7-a5c1-4a40036d3af9)

**Comment by gauss-clb** [comment]
I found out where the problem lies. 

In the `RewardManager`, you used `sequences_str` as llm_solution, which contained the prompt, but there is `<\think>` in my prompt, so this [line](https://github.com/agentica-project/rllm/blob/main/rllm/rewards/math_reward.py#L39) made a mistake.
```
sequences = torch.cat((valid_prompt_ids, valid_response_ids))
sequences_str = self.tokenizer.decode(sequences)

...

score = compute_score_fn(data_source=data_source, llm_solution=sequences_str, ground_truth=ground_truth)
```

**Comment by michaelzhiluo** [comment]
Gotcha, yes for our training we don't use system prompt! (as did Deepseek)

**Comment by gauss-clb** [comment]
> Gotcha, yes for our training we don't use system prompt! (as did Deepseek)

@michaelzhiluo Although I fixed this problem, the results of the two methods are still very different. This is very weird. Do you have any idea?

**Comment by michaelzhiluo** [comment]
Wow, this is a very nice find, I'll also consider this in future experiments...
